commit 54c72d5987ff9f3cf59529d5d4f5cf19eae3f695
Merge: 66bb0aa07797 7caa79917ad4
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Thu Aug 7 17:17:39 2014 -0700

    Merge tag 'mfd-for-linus-3.17' of git://git.kernel.org/pub/scm/linux/kernel/git/lee/mfd
    
    Pull MFD update from Lee Jones:
     "Changes to existing drivers:
       - checkpatch fixes throughout the subsystem
       - use Regmap to handle IRQs in max77686, extcon-max77693 and
         mc13xxx-core
       - use DMA in rtsx_pcr
       - restrict building on unsupported architectures on timberdale,
         cs5535
       - SPI hardening in cros_ec_spi
       - more robust error handing in asic3, cros_ec, ab8500-debugfs,
         max77686 and pcf50633-core
       - reorder PM runtime and regulator handing during shutdown in arizona
       - enable wakeup in cros_ec_spi
       - unused variable/code clean-up in pm8921-core, cros_ec, htc-i2cpld,
         tps65912-spi, wm5110-tables and ab8500-debugfs
       - add regulator handing into suspend() in sec-core
       - remove pointless wrapper functions in extcon-max77693 and
         i2c-cros-ec-tunnel
       - use cross-architecture friendly data sizes in stmpe-i2c, arizona,
         max77686 and tps65910
       - devicetree documentation updates throughout
       - provide power management support in max77686
       - few OF clean-ups in max77686
       - use manged resources in tps6105x
    
      New drivers/supported devices:
       - add support for s2mpu02 to sec-core
       - add support for Allwinner A32 to sun6i-prcm
       - add support for Maxim 77802 in max77686
       - add support for DA9063 AD in da9063
       - new driver for Intel PMICs (generic) and specifically Crystal Cove
    
      (Re-)moved drivers ==
       - move out keyboard functionality cros_ec ==> input/keyboard/cros_ec_keyb"
    
    * tag 'mfd-for-linus-3.17' of git://git.kernel.org/pub/scm/linux/kernel/git/lee/mfd: (101 commits)
      MAINTAINERS: Update MFD repo location
      mfd: omap-usb-host: Fix improper mask use.
      mfd: arizona: Only free the CTRLIF_ERR IRQ if we requested it
      mfd: arizona: Add missing handling for ISRC3 under/overclocked
      mfd: wm5110: Add new interrupt register definitions
      mfd: arizona: Rename thermal shutdown interrupt
      mfd: wm5110: Add in the output done interrupts
      mfd: wm5110: Remove non-existant interrupts
      mfd: tps65912-spi: Remove unused variable
      mfd: htc-i2cpld: Remove unused code
      mfd: da9063: Add support for AD silicon variant
      mfd: arizona: Map MICVDD from extcon device to the Arizona core
      mfd: arizona: Add MICVDD to mapped regulators for wm8997
      mfd: max77686: Ensure device type IDs are architecture agnostic
      mfd: max77686: Add Maxim 77802 PMIC support
      mfd: tps6105x: Use managed resources when allocating memory
      mfd: wm8997-tables: Suppress 'line over 80 chars' warnings
      mfd: kempld-core: Correct a variety of checkpatch warnings
      mfd: ipaq-micro: Fix coding style errors/warnings reported by checkpatch
      mfd: si476x-cmd: Remedy checkpatch style complains
      ...

diff --cc drivers/mfd/cros_ec_spi.c
index 8c1c7cc373f8,ac52e3653e90..588c700af39c
--- a/drivers/mfd/cros_ec_spi.c
+++ b/drivers/mfd/cros_ec_spi.c
@@@ -263,21 -266,18 +263,17 @@@ static int cros_ec_cmd_xfer_spi(struct 
  		dev_err(ec_dev->dev, "spi transfer failed: %d\n", ret);
  	}
  
- 	/* turn off CS */
+ 	/*
+ 	 * Turn off CS, possibly adding a delay to ensure the rising edge
+ 	 * doesn't come too soon after the end of the data.
+ 	 */
  	spi_message_init(&msg);
- 
- 	if (ec_spi->end_of_msg_delay) {
- 		/*
- 		 * Add delay for last transaction, to ensure the rising edge
- 		 * doesn't come too soon after the end of the data.
- 		 */
- 		memset(&trans, 0, sizeof(trans));
- 		trans.delay_usecs = ec_spi->end_of_msg_delay;
- 		spi_message_add_tail(&trans, &msg);
- 	}
+ 	memset(&trans, 0, sizeof(trans));
+ 	trans.delay_usecs = ec_spi->end_of_msg_delay;
+ 	spi_message_add_tail(&trans, &msg);
  
  	final_ret = spi_sync(ec_spi->spi, &msg);
 -	ktime_get_ts(&ts);
 -	ec_spi->last_transfer_ns = timespec_to_ns(&ts);
 +	ec_spi->last_transfer_ns = ktime_get_ns();
  	if (!ret)
  		ret = final_ret;
  	if (ret < 0) {